BEPs/BEP-467.md

```diff
@@ -0,0 +1,89 @@
+  Type: Information
+  Created: 2024-11-20
+  Author: TenArmor
+  Description: An out of protocol system to detect attack transaction for block builder
+  Discussions: https://forum.bnbchain.org/t/bep-idea-introduce-built-in-security-at-the-chain-level/3049
+</pre>
+
+# BEP-467: Transaction Inspection Zone (TIZ) for Attack Detection #
+
+## Status
+
+Draft / Proposal
+
+## Motivation
+
+This proposal introduces the **Transaction Inspection Zone (TIZ)**, a security layer designed for block builder to inspect all transactions within a block before they are published to validators. The TIZ will be operated on a dedicated server with a set of preconfigured rules to identify potential attacks and evaluate the impact of each transaction.
+
+By identifing and filtering out malicious or suspicious transactions, the TIZ will protect the network and reduce the risk of attacks that are currently plaguing the whole DEFI ecosystem.
+
+## Detailed Specification
+
+### Overview of TIZ Operation
+
+The TIZ operates as an **inspection layer** before transactions are sent to validators for inclusion in the blockchain. The primary function of the TIZ is to **detect malicious attack**. This is done by analyzing various information collected by simulating the transaction, TIZ consists of several components:
+
+- **Rule-Based Detection Engine**: This engine works as a general purpose engine to detect attack by matching common attack patterns. A set of rules designed by security experts will be applied to the transaction execution call trace, example of such rules include:
+  - Price manipulation.
+  - Invalid access control.
+  - Malicious priviledge changed.
+- **Impact Analysis Engine**: This is a dedicted engine/framework to support running test code before and after calling a contract, such dynamic inspection capability offers extreme flexibity to check the state changes brought by certain contract calls:
+  - Inspection code is evm compatible and can be deployed on-chain for transparency, it is protocol specific, and developed by protocol developers in their projects, something like invariant tests.
+  - Protocols register their inspection code through open API, the registration let protocols decide the exact condition as to when to run their inspection code, and define the precise inspection point during the execution of a transaction.
+- **Response Action Engine**: Transactions identified as potentially malicious are flagged for rejection or additional review, this engine allows user to carry out predefined actions.
+  - Proactive Vulnerability Response: in the event of an attack occuring it is crucial to act as quickly as possible, to do so protocols can registers actions for different situation beforehand, such as pausing protocol, disabling transfering of funds, etc.
+  - Builtin standard actions: rejecting transaction for the next block, notifying relavant parties, etc.
+
+Once inspected, the cleared transactions are forwarded back to the block builder.
```

Comment:
any design on validator side, like:
1. what if a validator does not integrate MEV?
2. or a validator do integrate with MEV but adopt local block eventually?

Reply:
Validators can integrate with this system independently.

Current design does not necessarily involve any changes to the consensus or Geth codebase. We believe such a design brings ease of adoption at the builder level and maintains the overall stability of the blockchain.